### PR TITLE
add tx into call chain

### PR DESCRIPTION
**Pull request by Schparky**
- There are a few unrelated minor changes mixed in, I got on a brief tangent of cleaning up lint warnings
- I am not convinced the mix of `*pop.Connection` and `context.Context` parameters is a good thing. I've gone back and forth on this, and haven't come to a solid conclusion. Cannot use `context.Context` everywhere because some functions are called places that do not have a context. Cannot use `*pop.Connection` everywhere unless a handful of functions are changed to accept the current user. Also, context is helpful for logging extras and calling domain.Logger with context.

### CODE REVIEW

File: application/models/meeting.go [hunk 1]
```diff
@@ -128,21 +128,21 @@ func (m *Meetings) FindOnDate(timeInFocus time.Time) error {
...
+	if err := getOrdered(m, Tx(ctx).Where("end_date >= ?", date)); err != nil {
 		return fmt.Errorf("error finding meeting with end_date before %s ... %s", date, err.Error())
 	}
 
 	return nil
 }
 
 // FindAfterDate finds the meetings that have a StartDate after the timeInFocus-date
-func (m *Meetings) FindAfterDate(timeInFocus time.Time) error {
+func (m *Meetings) FindAfterDate(tx *pop.Connection, timeInFocus time.Time) error {
```
Comment (Baggerone) [inline application/models/meeting.go:142]: Looks like this is only called from the tests. I see that it's different than FindOnOrAfterDate.

I wonder if they should have the same signature?

It looks like that is also true for FindOnDate.

**Review by Baggerone — APPROVED** [review]
Thanks for your work on this.

File: application/actions/authsocial.go [hunk 2]
```diff
@@ -132,7 +132,7 @@ func getSocialAuthOptions(authConfigs map[string]SocialAuthConfig) []authOption
 // For non invite based ... based on a User in the database with a SocialAuthProvider value
 func finishAuthRequestForSocialUser(c buffalo.Context, authEmail string) error {
 	var user models.User
-	if err := user.FindByEmail(authEmail); err != nil {
+	if err := user.FindByEmail(models.DB, authEmail); err != nil {
```
Comment (fillup) [inline application/actions/authsocial.go:135]: Is this right? Same on line 283.

File: application/internal/test/test.go [hunk 3]
```diff
@@ -175,7 +177,7 @@ func CreateFileFixture() models.File {
 		Name:    strconv.Itoa(rand.Int()) + ".gif",
 		Content: []byte("GIF89a"),
 	}
-	if err := f.Store(); err != nil {
+	if err := f.Store(models.DB); err != nil {
```
Comment (fillup) [inline application/internal/test/test.go:180]: is this right? It's a test so maybe using global connection is okay?

File: application/job/job.go [hunk 4]
```diff
@@ -42,11 +42,11 @@ func newThreadMessageHandler(args worker.Args) error {
 	}
 
 	var m models.Message
-	if err := m.FindByID(id, "SentBy", "Thread"); err != nil {
+	if err := m.FindByID(models.DB, id, "SentBy", "Thread"); err != nil {
```
Comment (fillup) [inline application/job/job.go:45]: Lots of `models.DB` in this file, is that right?

Comment (fillup) [reply application/job/job.go:45]: I'm thinking so if this runs as a background job and does not have a user context to pull a Tx from.

**Review by fillup — APPROVED** [review]

File: application/job/job.go (hunk 4 above)
Comment (Schparky) [reply application/job/job.go:45]: Right

File: application/internal/test/test.go (hunk 3 above)
Comment (Schparky) [reply application/internal/test/test.go]: Probably, but I'll pass in the connection anyway.

File: application/actions/authsocial.go (hunk 2 above)
Comment (Schparky) [reply application/actions/authsocial.go]: No, good find.

File: application/models/meeting.go (hunk 1 above)
Comment (Schparky) [reply application/models/meeting.go:142]: Changed, thanks.

**Pull request merged**